lms/static/js/csv_viewer/data/actions/csvFetcher.js

```diff
@@ -0,0 +1,49 @@
+import {csvViewerActions} from './constants';
+import Papa from 'papaparse';
+
+const fetchCSVDataSuccess = csvData => {
+    return {
+        type: csvViewerActions.fetch.SUCCESS,
+        csvData,
+    };
+};
+
+const fetchCSVDataFailure = error => {
+    return {
+        type: csvViewerActions.fetch.FAILURE,
+        error,
+    };
+};
+
+const fetchCSVData = csvFileUrl => (dispatch) => {
+    fetch(csvFileUrl)
+        .then(response => {
+            if (response.ok) {
+                return response.text()
+            } else {
+                throw new Error();
+            }
+        })
```

Comment:
It seems papaparse [is able to download remote files by itself](https://www.papaparse.com/#remote-files), maybe that simplify things here?

Reply:
Sure will do, that does seem cleaner.